### PR TITLE
CAS-370 Add prison release type to temporary accommodation application submit

### PR DESCRIPTION
**Pull request by muhammad-elabdulla**
This [PR CAS-370](https://dsdmoj.atlassian.net/browse/CAS-370) is to add prison release types when submit temporary accommodation application

### CODE REVIEW

File: src/main/resources/db/migration/all/20240513120304__add_application_prison_release_types_table.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE application_prison_release_types (
```
Comment (davidatkinsuk) [inline src/main/resources/db/migration/all/20240513120304__add_application_prison_release_types_table.sql:1]: I think this should maybe be called 'temporary_accommodation_application_prison_release_types' as it's only applicable to cas3 application?

**Review by davidatkinsuk — APPROVED** [review]
Just one comment from me

File: src/main/resources/db/migration/all/20240513120304__add_application_prison_release_types_table.sql (hunk 1 above)
Comment (muhammad-elabdulla) [reply src/main/resources/db/migration/all/20240513120304__add_application_prison_release_types_table.sql]: Changed. I named it as application_prison_release_types if another CAS need to use it in the future

Comment (davidatkinsuk) [reply src/main/resources/db/migration/all/20240513120304__add_application_prison_release_types_table.sql]: I was considering that but noticed we had a simple table for CAS1 that was prefixed with approved_premises_applicaitons. we can always rename (and move relationship to the common application entity in code) if needed by other CASs

**Pull request merged**